homeassistant/components/vicare/__init__.py

```diff
@@ -82,15 +72,7 @@ def setup_vicare_api(hass: HomeAssistant, entry: ConfigEntry) -> None:
             "Found device: %s (online: %s)", device.getModel(), str(device.isOnline())
         )
 
-    # Currently we only support a single device
-    device_list = vicare_api.devices
-    device = device_list[0]
-    hass.data[DOMAIN][entry.entry_id][VICARE_DEVICE_CONFIG_LIST] = device_list
-    hass.data[DOMAIN][entry.entry_id][VICARE_DEVICE_CONFIG] = device
-    hass.data[DOMAIN][entry.entry_id][VICARE_API] = getattr(
-        device,
-        HEATING_TYPE_TO_CREATOR_METHOD[HeatingType(entry.data[CONF_HEATING_TYPE])],
-    )()
+    hass.data[DOMAIN][entry.entry_id][DEVICE_CONFIG_LIST] = vicare_api.devices
```

Comment:
```suggestion
    hass.data[DOMAIN][entry.entry_id][DEVICE_CONFIG_LIST] = [(device, get_device(entry, device_config)) for device in vicare_api.devices]
```

Please adopt all platforms

Reply:
What type will be stored in DEVICE_CONFIG_LIST now?

Reply:
It will be a tuple of both values